### PR TITLE
Replace ckanext-keycloak by ckanext-oidc-pkce

### PR DESCRIPTION
**Pull request by brunopacheco1**

### CODE REVIEW

File: ckan/Dockerfile [hunk 1]
```diff
@@ -15,8 +15,8 @@ RUN  pip3 install -e git+https://github.com/GenomicDataInfrastructure/gdi-userpo
```
Comment (hcvdwerf) [inline ckan/Dockerfile:15]: Is this  file also not obsolete now, because this project is focussing on local development?

**Review by hcvdwerf — CHANGES_REQUESTED** [review]
2023-12-14 11:51:34,419 WARN  [io.agroal.pool] (agroal-11) Datasource '<default>': FATAL: password authentication failed for user "keycloakdbuser"
gdi-userportal-ckan-docker-keycloak-1                 | 2023-12-14 11:51:34,420 WARN  [org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator] (JPA Startup Thread) HHH000342: Could not obtain connection to query metadata: java.lang.NullPointerException: Cannot invoke "org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(java.sql.SQLException, String)" because the return value of "org.hibernate.resource.transaction.backend.jta.internal.JtaIsolationDelegate.sqlExceptionHelper()" is null
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.resource.transaction.backend.jta.internal.JtaIsolationDelegate.doTheWork(JtaIsolationDelegate.java:186)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.resource.transaction.backend.jta.internal.JtaIsolationDelegate.lambda$delegateWork$1(JtaIsolationDelegate.java:75)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.resource.transaction.backend.jta.internal.JtaIsolationDelegate.doInSuspendedTransaction(JtaIsolationDelegate.java:107)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.resource.transaction.backend.jta.internal.JtaIsolationDelegate.delegateWork(JtaIsolationDelegate.java:72)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator.getJdbcEnvironmentUsingJdbcMetadata(JdbcEnvironmentInitiator.java:279)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator.initiateService(JdbcEnvironmentInitiator.java:193)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator.initiateService(JdbcEnvironmentInitiator.java:69)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.boot.registry.internal.StandardServiceRegistryImpl.initiateService(StandardServiceRegistryImpl.java:119)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.service.internal.AbstractServiceRegistryImpl.createService(AbstractServiceRegistryImpl.java:264)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.service.internal.AbstractServiceRegistryImpl.initializeService(AbstractServiceRegistryImpl.java:239)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.service.internal.AbstractServiceRegistryImpl.getService(AbstractServiceRegistryImpl.java:216)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.engine.jdbc.internal.JdbcServicesImpl.configure(JdbcServicesImpl.java:52)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.boot.registry.internal.StandardServiceRegistryImpl.configureService(StandardServiceRegistryImpl.java:125)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.service.internal.AbstractServiceRegistryImpl.initializeService(AbstractServiceRegistryImpl.java:248)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.service.internal.AbstractServiceRegistryImpl.getService(AbstractServiceRegistryImpl.java:216)
gdi-userportal-ckan-docker-keycloak-1                 |         at org.hibernate.boot.internal.SessionFactoryOptionsBuilder.<init>(SessionFactoryOptionsBuilder.java:273)
gdi-userportal-ckan-docker-keycloak-1                 |         at io.quarkus.hibernate.orm.runtime.recording.PrevalidatedQuarkusMetadata.buildSessionFactoryOptionsBuilder(PrevalidatedQuarkusMetadata.java:70)
gdi-userportal-ckan-docker-keycloak-1                 |         at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:81)
gdi-userportal-ckan-docker-keycloak-1                 |         at io.quarkus.hibernate.orm.runtime.FastBootHibernatePersistenceProvider.createEntityManagerFactory(FastBootHibernatePersistenceProvider.java:74)
gdi-userportal-ckan-docker-keycloak-1                 |         at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:80)
gdi-userportal-ckan-docker-keycloak-1                 |         at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:55)
gdi-userportal-ckan-docker-keycloak-1                 |         at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:156)
gdi-userportal-ckan-docker-keycloak-1                 |         at io.quarkus.hibernate.orm.runtime.JPAConfig$1.run(JPAConfig.java:64)
gdi-userportal-ckan-docker-keycloak-1                 |         at java.base/java.lang.Thread.run(Thread.java:840)
gdi-userportal-ckan-docker-keycloak-1                 |

File: ckan/Dockerfile (hunk 1 above)
Comment (brunopacheco1) [reply ckan/Dockerfile:15]: Yes and no, I would say. 

In theory, the development life cycle includes code, test, build, publish. The artefact delivered by this repo is a docker image. Any suggestion on how to blend the two docker files together?

**Comment by brunopacheco1** [comment]
> 2023-12-14 11:51:34,419 WARN [io.agroal.pool] (agroal-11) Datasource '': FATAL: password authentication failed for user "keycloakdbuser" gdi-userportal-ckan-docker-keycloak-1 | 2023-12-14 11:51:34,420 WARN [org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator] (JPA Startup Thread) HHH000342: Could not obtain connection to query metadata: java.lang.NullPointerException: Cannot invoke "org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(java.sql.SQLException, String)" because the return value of "org.hibernate.resource.transaction.backend.jta.internal.JtaIsolationDelegate.sqlExceptionHelper()" is null gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.resource.transaction.backend.jta.internal.JtaIsolationDelegate.doTheWork(JtaIsolationDelegate.java:186) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.resource.transaction.backend.jta.internal.JtaIsolationDelegate.lambda$delegateWork$1(JtaIsolationDelegate.java:75) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.resource.transaction.backend.jta.internal.JtaIsolationDelegate.doInSuspendedTransaction(JtaIsolationDelegate.java:107) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.resource.transaction.backend.jta.internal.JtaIsolationDelegate.delegateWork(JtaIsolationDelegate.java:72) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator.getJdbcEnvironmentUsingJdbcMetadata(JdbcEnvironmentInitiator.java:279) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator.initiateService(JdbcEnvironmentInitiator.java:193) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator.initiateService(JdbcEnvironmentInitiator.java:69) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.boot.registry.internal.StandardServiceRegistryImpl.initiateService(StandardServiceRegistryImpl.java:119) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.service.internal.AbstractServiceRegistryImpl.createService(AbstractServiceRegistryImpl.java:264) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.service.internal.AbstractServiceRegistryImpl.initializeService(AbstractServiceRegistryImpl.java:239) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.service.internal.AbstractServiceRegistryImpl.getService(AbstractServiceRegistryImpl.java:216) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.engine.jdbc.internal.JdbcServicesImpl.configure(JdbcServicesImpl.java:52) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.boot.registry.internal.StandardServiceRegistryImpl.configureService(StandardServiceRegistryImpl.java:125) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.service.internal.AbstractServiceRegistryImpl.initializeService(AbstractServiceRegistryImpl.java:248) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.service.internal.AbstractServiceRegistryImpl.getService(AbstractServiceRegistryImpl.java:216) gdi-userportal-ckan-docker-keycloak-1 | at org.hibernate.boot.internal.SessionFactoryOptionsBuilder.(SessionFactoryOptionsBuilder.java:273) gdi-userportal-ckan-docker-keycloak-1 | at io.quarkus.hibernate.orm.runtime.recording.PrevalidatedQuarkusMetadata.buildSessionFactoryOptionsBuilder(PrevalidatedQuarkusMetadata.java:70) gdi-userportal-ckan-docker-keycloak-1 | at io.quarkus.hibernate.orm.runtime.boot.FastBootEntityManagerFactoryBuilder.build(FastBootEntityManagerFactoryBuilder.java:81) gdi-userportal-ckan-docker-keycloak-1 | at io.quarkus.hibernate.orm.runtime.FastBootHibernatePersistenceProvider.createEntityManagerFactory(FastBootHibernatePersistenceProvider.java:74) gdi-userportal-ckan-docker-keycloak-1 | at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:80) gdi-userportal-ckan-docker-keycloak-1 | at jakarta.persistence.Persistence.createEntityManagerFactory(Persistence.java:55) gdi-userportal-ckan-docker-keycloak-1 | at io.quarkus.hibernate.orm.runtime.JPAConfig$LazyPersistenceUnit.get(JPAConfig.java:156) gdi-userportal-ckan-docker-keycloak-1 | at io.quarkus.hibernate.orm.runtime.JPAConfig$1.run(JPAConfig.java:64) gdi-userportal-ckan-docker-keycloak-1 | at java.base/java.lang.Thread.run(Thread.java:840) gdi-userportal-ckan-docker-keycloak-1 |

Let's get together tomorrow, if you are available. It seems to be environment configuration issue.

File: ckan/Dockerfile (hunk 1 above)
Comment (brunopacheco1) [reply ckan/Dockerfile:15]: Peharps, to not block this PR, I suggest we open another issue to review this later. What do you think, @hcvdwerf ?

**Review by hcvdwerf — APPROVED** [review]

**Pull request merged**